platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblems.java

```diff
@@ -24,13 +27,16 @@
 import java.util.Collections;
 import java.util.List;
 
+import static org.gradle.api.problems.internal.DefaultProblemCategory.GRADLE_CORE_NAMESPACE;
+
 @ServiceScope(Scope.BuildTree.class)
 public class DefaultProblems implements InternalProblems {
 
     private final CurrentBuildOperationRef currentBuildOperationRef;
     private final ProblemEmitter emitter;
     private final List<ProblemTransformer> transformers;
     private final InternalProblemReporter internalReporter;
+    private final Multimap<Throwable, Problem>  problems = Multimaps.synchronizedMultimap(HashMultimap.<Throwable, Problem>create());
```

Comment:
🤔 I find the name of this variable misleading, as I would expect `problems` to contain what? All the problems? 

If I were presented with this, I probably could deduce that this is not the case, as `Throwable` is not present in all problems, so this is more of a failure mapping.